src/ServiceControl.Persistence.RavenDB/RavenEmbeddedHostedService.cs

```diff
@@ -0,0 +1,34 @@
+namespace ServiceControl.Persistence.RavenDB;
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+class RavenEmbeddedHostedService(RavenPersisterSettings configuration)
+    : IHostedService, IConnectionStringProvider
+{
+    EmbeddedDatabase embeddedDatabase;
+
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        embeddedDatabase = EmbeddedDatabase.Start(configuration);
+        return Task.CompletedTask;
+    }
+
+    public Task StopAsync(CancellationToken cancellationToken)
+    {
+        embeddedDatabase.Dispose();
+        return Task.CompletedTask;
+    }
+
+    public async Task<string> GetConnectionString()
+    {
+        while (embeddedDatabase == null)
```

Comment:
Not sure if this is the best way, but the idea is to wait till the server is ready